### DISCUSSION TITLE
Reverse Appending

### DISCUSSION THREAD

**Discussion by Jay123210599** [ImageMagick/ImageMagick#7382] [Help]
### ImageMagick version

7.1.1-27

### Operating system, version and so on

Windows 11

### Description

I have learned to merge files together, vertically and horizontally. Now, how do I reverse its effects, and/or losslessly separate them without losing any quality?

![merger](https://github.com/ImageMagick/ImageMagick/assets/141977822/874de08f-e6f2-42bd-bd37-9386ee007a42)

https://github.com/ImageMagick/ImageMagick/discussions/7175
https://github.com/ImageMagick/ImageMagick/discussions/7335
https://github.com/ImageMagick/ImageMagick/discussions/7361
https://github.com/ImageMagick/ImageMagick/discussions/7363
https://github.com/ImageMagick/ImageMagick/discussions/7364

**Comment by fmw42** [comment]
If you know the original sizes of the images that you appended, then you can simply use -crop WxH+X+Y +repage on your PNG image twice to make two more images.  See the -crop operator at https://imagemagick.org/Usage/crop/#crop.  If both images are the same height (since you stacked them vertically), you should be able to crop them once and make two output PNGs.  See https://imagemagick.org/Usage/crop/#crop_tile

There is no way to get JPGs merged and then cropped back to their exact originals, since JPG uses a lossy compression, even at -quality 100.

**Reply by Jay123210599** [reply]
In the image above, there were two images that were both at 1920x1080. Could you use that as example?

**Comment by snibgo** [comment]
There are many ways to do this. If the height is a multiple of 2, then I suggest `-crop 1x2@` which chops the input into 2 equal-sized pieces. Like this:
```
magick in.png -crop 1x2@ out.png
```
